packages/amplify-dotnet-function-runtime-provider/amplify-plugin.json

```diff
@@ -11,8 +11,8 @@
     },
     "runtimes": [
       {
-        "name": ".NET Core 3.1",
-        "value": "dotnetcore3.1"
+        "name": ".NET 6",
```

Comment:
Can we support both runtimes and can detect runtime based on Cx project ?

Also we can warn to Cx that ```NET Core 3.1``` is getting deprecated.

Reply:
How about this ?
![image](https://user-images.githubusercontent.com/5849952/206769694-9344934f-df5d-492a-bf88-ce34ea6a0a3b.png)

Reply:
btw. this `amplify-plugin.json`'s `runtimes` section doesn't seem to be used for anything. I'd keep just .net6 here and not advertise deprecated tech. @jhockett @akshbhu do you see any reasons we should keep net core here? we're removing ability to generate new functons from netcore templates.

Reply:
yeah I agree , it's getting deprecated , we shouldn't allow to create ```dotnetcore31 ``` runtime.

Reply:
rebased to dev and added migration test.